src/main/com/fulcrologic/rad/form.cljc

```diff
@@ -717,14 +719,7 @@
 
 (def global-events
   {:event/exit
-   {::uism/handler (fn [{::uism/keys [event-data fulcro-app] :as env}]
-                     (let [route (::new-route event-data)]
-                       (cond
-                         (and (= :back route) (history/history-support? fulcro-app)) (history/back! fulcro-app)
-                         (and (nil? route) (history/history-support? fulcro-app)) (history/back! fulcro-app)
-                         (vector? route) (dr/change-route! fulcro-app route))
-                       (uism/exit env)))}
-
+   {::uism/handler (fn [env] (uism/exit env))}
```

Comment:
This `:event/exit` was the culprit behind the flicker. It is called in two situations.

1. User clicks the done button, in which case we want to invoke `history/back!`
2. When the user routes away from an open form (clicks side menu), in which case we don't want to do anything since history has already been updated to the correct value by user-driven routing.

So I moved the `history/back!` call to the `exit-form` handler which is tied to the done button. This was happening for all entities but we weren't noticing for some because of a timing issue. By the time the history API went backwards the prior-route had been updated to what the user routed to.